src/main.py

```diff
@@ -80,22 +80,34 @@
 # when you use a file as a program, but not if you import
 # the file for use as a module.
 if __name__ == '__main__':
+
     if len(sys.argv) > 3 or len(sys.argv) < 2:
         sys.exit(1)
-    if len(sys.argv) in [2,3]:
-        inFile = sys.argv[1]
+
     if len(sys.argv) == 3:
+        inFile = open(sys.argv[1], 'r')
         outFile = open(sys.argv[2],'w')
-    
 
-    with open(inFile,'r') as file:
-            lines = file.readlines() #lines()
+    if len(sys.argv) == 2:
+        if sys.stdin.isatty() == False:  # False if stdin is connected.
```

Comment:
I'm not sure what you are doing here, but I am sure it is wrong. The `isatty()` method checks if standard input is a terminal, but there is nothing in this exercise that says we should treat terminates different from other types of input pipes.

Consider this simple program:

```python
import sys
print(sys.stdin.isatty(), sys.stdin.read())
```

that I will put in `foo.py`. It tells us if the standard in is a terminal and what we can read from it.

If we call it like this:

```sh
bash$ python3 foo.py
hello
True hello
```

we connect the terminal to its standard input, so we can write hello and close the pipe with Ctrl-D. The result is as shown.

We could also pipe the output of one command into `foo.py` to get the input from there:

```sh
bash$ echo hello | python3 foo.py
False hello

```

or we could make a file and redirect that:

```sh
bash$ echo hello > hello.txt
bash$ python3 foo.py < hello.txt
False hello
```

The first case isn't a special case for this program, nor should it be for this exercise. Checking if you are in the first of the three situations is not necessary. You just need to use `stdin` when you don't have a filename as an argument.